nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublishKafkaRecord_2_6.java

```diff
@@ -124,6 +125,14 @@
         "Interprets the <Partition> property as Expression Language that will be evaluated against each FlowFile. This Expression will be evaluated once against the FlowFile, " +
             "so all Records in a given FlowFile will go to the same partition.");
 
+    static final AllowableValue MESSAGE_PROPERTY_KEY_RESOLVER_SELECTION = new AllowableValue("MessageValuePropertyResolver",
```

Comment:
Rather than a naming convention of MessageValuePropertyResolver, RecordPathResolver, etc. remember: this is a user-facing configuration option. So would stick with more natural language: "Use Field in Record", "Use RecordPath", "Use Literal Value" (or "Use Expression Language")